Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -171,7 +171,9 @@ class MessageManager: EngineWebDelegate {
                 logger.logWithModuleTag("Dismissing from action: \(action)", level: .info)
                 inAppMessageManager.dispatch(action: .dismissMessage(message: currentMessage, viaCloseAction: true))
             case "loadPage":
-                if let page = url.queryParameters?["url"],
+                // Encode '#' in url action string and creates encoded URL to handle fragments
+                let encodedUrl = URL(string: action.percentEncode(character: "#")) ?? url
```

Comment:
wonder why don't we encode the whole URL? instead of separating out fragment? 

```
if let page = url.queryParameters?["url"]
let encodedPage = page.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
let pageUrl = URL(string: page)
```

so instead of encoding, just `#` we encode everything except that allowed and I would hope the webview opens it just fine.

Reply:
That's what I've been considering as well. However, I opted to keep the impact minimal and decided to encode only `#` for following reasons:

- The link is provided directly by customers, so I wanted to avoid making any unnecessary modifications. While I agree that browsers and most apps should not be affected by encoding, I'm unsure if encoding full URL might impact any specific customer links.
- On Android, [we don't directly encode full URL either](https://github.com/customerio/customerio-android/blob/a402e9645df501bba8ff5d32ee423d8f818635e4/messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistView.kt#L88-L94), but `UrlQuerySanitizer` respects fragments (`#`) without encoding them, whereas iOS drops them entirely. This change simply makes iOS behavior a bit closer to Android, while relying on default `URL` class for encoding the rest.
- Based on my testing, this code is only used when utilizing "Link to web page" action in legacy in-app message editor. With deep link option and new editor, links are handled differently. So overall impact of this change is very low.

I'm open to encoding full URL as well, and I agree it should work fine for most browsers. However, I'm slightly inclined toward encoding only `#` to minimize potential risks since the url is customer provided, and I cannot fully verify all possible impacts. By encoding only `#`, the behavior remains closer to Android while making minimal changes to current implementation.

Reply:
This sounds reasonable if the android counter part is not fully encoding it either. I just want to avoid the case, where if someone adds another character allowed by URL and we end up filtering it out just like we did for `#`. 

One thing to note is, even though android doesn't encode it, `URI.parse` does is able to distinguish the following 

```
foo://example.com:8042/over/there?name=ferret#nose
\_/   \______________/\_________/ \_________/ \__/
 |           |            |            |        |
scheme    authority      path        query   fragment
```

you can see there is usage of `?` `=` as well along with `#`, I assume the iOS counter part won't be able to handle these?